common/txmgr/types/tx_store.go

```diff
@@ -84,7 +84,7 @@ type TransactionStore[
 	SaveSentAttempt(timeout time.Duration, attempt *TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], broadcastAt time.Time) error
 	SetBroadcastBeforeBlockNum(blockNum int64, chainID CHAIN_ID) error
 	UpdateBroadcastAts(now time.Time, etxIDs []int64) error
-	UpdateTxAttemptInProgressToBroadcast(etx *Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], attempt TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], NewAttemptState TxAttemptState, incrNextSequenceCallback QueryerFunc, qopts ...pg.QOpt) error
+	UpdateTxAttemptInProgressToBroadcast(etx *Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], attempt TxAttempt[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE], NewAttemptState TxAttemptState, incrementSeqFunc func(address ADDR) error) error
```

Comment:
Probably no need to take as input the incrementSeqFunc here.
Instead, let this function return success to the broadcaster.
The broadcaster then, after seeing the success, can directly increase the sequence there itself.

This way, we keep this TxStore interface simpler.

Reply:
I originally considered this but thought there could have been issues with removing the increment from the atomic transaction. Like if an error occurred when incrementing the sequence, we'd want to revert the changes to tx too.